pkg/antctl/raw/test/command.go

```diff
@@ -0,0 +1,493 @@
+			Port:    80,
+			Image:   "k8s.gcr.io/e2e-test-images/agnhost:2.31",
+			Command: []string{"/agnhost", "netexec", "--http-port=80"},
+			Affinity: &corev1.Affinity{
+				PodAntiAffinity: &corev1.PodAntiAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+						{
+							LabelSelector: &metav1.LabelSelector{
+								MatchExpressions: []metav1.LabelSelectorRequirement{
+									{Key: "name", Operator: metav1.LabelSelectorOpIn, Values: []string{clientDeploymentName}},
+								},
+							},
+							TopologyKey: "kubernetes.io/hostname",
+						},
+					},
+				},
+			},
+			Labels: map[string]string{"app": echoOtherNodeDeploymentName},
+		})
+
+		_, err = k.clients.destination.CreateDeployment(ctx, k.namespace, echoOtherNodeDeployment, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("unable to create deployment %s: %s", echoOtherNodeDeploymentName, err)
+		}
+	}
+
+	return nil
+}
+
+func (k *k8sConnectivityParams) waitForDeploymentsReady(ctx context.Context, client k8sClientOperations, deployments []string, interval, timeout time.Duration) error {
+	for _, deployment := range deployments {
+		k.Log("[%s] Waiting for deployment %s to become ready...", client.ClusterName(), deployment)
+		err := wait.PollUntilContextTimeout(ctx, interval, timeout, false, func(ctx context.Context) (bool, error) {
+			ready, err := client.DeploymentIsReady(ctx, k.namespace, deployment)
+			if err != nil {
+				return false, fmt.Errorf("error checking readiness of deployment %s: %w", deployment, err)
+			}
+			return ready, nil
+		})
+		if err != nil {
+			return fmt.Errorf("waiting for deployment %s to become ready has been interrupted: %w", deployment, err)
+		}
+		k.Log("[%s] Deployment %s is ready.", client.ClusterName(), deployment)
+	}
+	return nil
+}
+
+func (k *k8sConnectivityParams) validateDeployment(ctx context.Context) error {
+	var err error
+
+	srcDeployments, dstDeployments := k.deploymentList()
+	if err := k.waitForDeploymentsReady(ctx, k.clients.source, srcDeployments, time.Second, podReadyTimeout); err != nil {
+		return err
+	}
+	if err := k.waitForDeploymentsReady(ctx, k.clients.destination, dstDeployments, time.Second, podReadyTimeout); err != nil {
+		return err
+	}
+
+	k.clientPods, err = k.client.ListPods(ctx, k.namespace, metav1.ListOptions{LabelSelector: "kind=" + kindClientName})
+	if err != nil {
+		return fmt.Errorf("unable to list client pods: %s", err)
+	}
+
+	k.echoPods = map[string]string{}
+	for _, client := range k.clients.clients() {
+		echoPods, err := client.ListPods(ctx, k.namespace, metav1.ListOptions{LabelSelector: "kind=" + kindEchoName})
+		if err != nil {
+			return fmt.Errorf("unable to list echo pods: %s", err)
+		}
+		for _, echoPod := range echoPods.Items {
+			k.echoPods[echoPod.Name] = echoPod.Status.PodIP
+		}
+	}
+	fmt.Fprintf(k.Writer, "Deployment is validated\n")
+	return nil
+}
+
+func (k *k8sConnectivityParams) Log(format string, a ...interface{}) {
```

Comment:
If you always use `os.Stdout` as the Writer to initialize this command, Maybe remove the writer field and call `fmt.Fprintf(os.Stdout, format+"\n", a...)` directly.  I feel there might be no need to provide multiple ways to record the logs.  

@antoninbas  what's your suggestion?